### PR TITLE
Add HIP build to Jenkins CI

### PR DESCRIPTION
**Pull request by dalg24**

### CODE REVIEW

**Comment by dalg24** [comment]
This is ready for review/merge

File: .jenkins [hunk 1]
```diff
@@ -9,6 +9,36 @@ pipeline {
...
+                            args '-v /tmp/ccache.kokkos:/tmp/ccache --device=/dev/kfd --device=/dev/dri --security-opt seccomp=unconfined --group-add video'
+                        }
+                    }
+                    steps {
+                        sh 'ccache --zero-stats'
+                        sh '''rm -rf build && mkdir -p build && cd build && \
+                              cmake \
+                                -DCMAKE_BUILD_TYPE=Debug \
+                                -DCMAKE_CXX_COMPILER_LAUNCHER=ccache \
+                                -DCMAKE_CXX_COMPILER=hipcc \
```
Comment (crtrott) [inline .jenkins]: Should we set explicitly Kokkos_ARCH_VEGA=ON?

Comment (dalg24) [reply .jenkins]: Probably so

File: scripts/docker/Dockerfile.hipcc [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        kmod \
+        wget \
+        ccache \
+        && \
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/*
+
+ENV PATH=/opt/rocm/bin:$PATH
+
+ARG CMAKE_VERSION=3.10.3
```
Comment (Rombur) [inline scripts/docker/Dockerfile.hipcc]: Any reason why you take an older version?

Comment (dalg24) [reply scripts/docker/Dockerfile.hipcc]: Kokkos supports 3.10 but we do have other builds with that version so I can bump it if you like.

Comment (crtrott) [reply scripts/docker/Dockerfile.hipcc]: its fine for now.

**Pull request merged**